File: IT/Eccezioni/Esercizi/Java/Esercizio2.java
/* file Esercizio2.java */

public class Esercizio2 {
	public static void main(String [] args) {
		String [] arr = {"United", "States", "", "America"};

		for (String s : arr) {
			try {
				System.out.print(s.charAt(0));
			}
			catch (StringIndexOutOfBoundsException e) {}
		}

		System.out.print(System.lineSeparator());
		System.out.println("Fine del programma.");
	}
}

File: IT/Eccezioni/Esercizi/C++/esercizio2.cpp
/*
file esercizio2.cpp
compilare con: g++ -std=c++11 esercizio2.cpp
*/

#include <iostream>
#include <array>
#include <stdexcept>

using std::cout;
using std::endl;
using std::string;
using std::array;
using std::out_of_range;

int main()
{
	array<string, 4> arr = {"United", "States", "", "America"};

	for (const string &s : arr) {
		try {
			cout << s.at(0);
		}
		catch (out_of_range& e) {}
	}

	cout << endl;
	cout << "Fine del programma." << endl;
}

